### DISCUSSION TITLE
How do I get nested data into the database(POST request)...

### DISCUSSION THREAD

**Discussion by ChrisB007** [prisma/prisma#10427] [Q&A] [answered]
Hello guys, 
I am a noob working on a PrismaORM project. On the project, I have set up my schema/models, setup and CRUD my routes, but when I tried using Postman to make a POST request with sample data, I am getting this error:

```
PrismaClientValidationError: 
Invalid `prisma.template.create()` invocation in
'path-to-my-file'

  40   option_5,
  41   option_6,
  42 } = req.body;
→ 43 const template = await prisma.template.create({
       data: {
         title: 'Top 5 Questions',
         email: 'test@testemail.com',
         child_questions: undefined,
         answers: {
         ~~~~~~~
           option_1: undefined,
           option_2: undefined,
           option_3: undefined,
           option_4: undefined,
           option_5: undefined,
           option_6: undefined
         }
       }
     })

Unknown arg `answers` in data.answers for type TemplateCreateInput. Did you mean `title`? Available args:
type TemplateCreateInput {
  title: String
  email: String
  questions?: QuestionCreateNestedManyWithoutTemplateInput
}

```

Here's what my Prisma.Schema file look like:

```
// schema.model file

model Template {
  id        Int        @id @default(autoincrement())
  title     String
  somelists Somelist[ ]            //Has many'items'
}

model Somelist {
  id              Int       @id @default(autoincrement())
  owner_id        Int       @unique
  campaign_id     Int       @unique
  questionId      Int       @unique
  child_questions  String
  someotherlists         Someotherlist[ ]    //Has many 'Items'
  correct_answer  String
  createdAt       DateTime  @default(now())
  updatedAt       DateTime  @default(now())
  Template        Template? @relation(fields: [TemplateId], references: [id])
  TemplateId      Int?
}



model Someotherlist {
  id         Int      @id @default(autoincrement())
  option_1   String?
  option_2   String?
  option_3   String?
  option_4   String?
  option_5   String?
  option_6   String?
  Somelist  Somelist? @relation(fields: [somelistId], references: [id])
   SomelistId Int
  createdAt  DateTime @default(now())
  updatedAt  DateTime @updatedAt
}

```

Here's what my POST route look like:

```
router.post("/templates", async (req, res, next) => {
  try {
    const {
      title,
      email,
      child_questions,
      option_1,
      option_2,
      option_3,
      option_4,
      option_5,
      option_6,
    } = req.body;
    const template = await prisma.template.create({
      data: {
        title,
        email,
        child_questions,
        answers: { option_1, option_2, option_3, option_4, option_5, option_6 },
      },
    });
    res.status(200).json(template);
  } catch (error) {
    console.log(error);
    next(error);
  }
});

```

Here is my test data:

```
{
    "title": "Top 5 Questions",
    "questions": {
    "child_questions": "This is some question to be answered?",
    "someotherlists ": {
      "option_1,": "Some answer 1",
      "option_2,": "Some answer 2",
      "option_3,": "Some answer 3",
      "option_4": "Some answer 4"
      "option_5": "Some answer 5"
      "option_6": "Some answer 6"
      },
  }
}

```

Expected Output:

```

{
    id: 1,
    title: "Some Title",
    questions: [
      {
        question: "How do you feel about solving this issue?",
        answers: [
          "Doing it for charity",
          "Was bored doing other things",
          "I help folks regularly",
          "Curious to learn more",
        ],
      },
}


```

- Thanks in advance

**Comment by NickFoden** [comment, marked as answer]
When you create this "template" item it takes 1 field -  The `title`, the other value somelists needs to be connected and cannot not be created as argument to "data"

```
model Template {
  id      // this is set to be auto generated
  title     String
  somelists // this you have to create/connect
  }
  ```
  
  So this template create won't work, because these fields don't exist on a `template`, only the title exists on a `template`:
  
  ```
   const template = await prisma.template.create({
      data: {
        title,
        email,
        child_questions,
        answers: { option_1, option_2, option_3, option_4, option_5, option_6 },
      },
    });
  ```
  
  Will need to create the `somelist` and link it, but then you have 2 levels/models to link up to create 1 `template`, and so you have to make the children of `somelist` first. The `Someotherlist` is needed in order to create this `somelist` so first we make a `SomeOtherList`
  
  
###   pseudo code below here as example 

I am thinking this could work (will have to verify and might not work as copy paste)
  
  First create a `someotherlist`
  
  ```
  const newSomeOtherList = await prisma.someotherlist.create({
    data:{
          option_1: option_1
          option_2:   option_2
          option_3:   option_3
          option_4:   option_4
          option_5:   option_5
          option_6:   option_6
  }})
  ```
  now use this new item and the id value to connect/create a new `someList`
  
  ```
  const newSomeList = await prisma.someList.create({
  data:{
    owner_id: owner_id,
    campaign_id:campaign_id, 
    questionId: questionId,
    child_questions: child_questions,
    correct_answer: correct_answer, 
    someotherlists:{
          connect: {
                id: newSomeOtherList.id
           }
      }
})
   ```
   
  And then now we are ready to create a new template and connect it to this `newSomeList` we have created. 
  
  ```
   const template = await prisma.template.create({
      data: {
        title,
        somelists:{
              connect: {
                   id: newSomeList.id,
             }
          },
    });

  ```

 There is a more efficient way then these "3" create calls to get 1 `template` created, as you can do a create within where i have `connect`s
 
  https://www.prisma.io/docs/concepts/components/prisma-client/crud#create-a-single-record-using-generated-types

 but at the moment I think this could be a clear way to get started with this current shape working, and so you can follow what is happening (we have to create the other items in order to link them up with the template) and then also you might update the shape of your quizzes as you go to remove some of these fields and maybe flatten out the relations a bit think can just be 2 models maybe. And suggest to rename these models so is more clear. "Quiz", "QuizQuestions" type naming instead of SomeList and SomeOtherList. I think the current naming is bit tougher for other people to review your code and follow the logic.

**Reply by ChrisB007** [reply]
Thank you. This make sense

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
